drv/lpc55-sprot-server/src/handler.rs

```diff
@@ -50,213 +54,115 @@ impl Handler {
             sprocket: crate::handler::sprockets::init(),
             update: Update::from(UPDATE_SERVER.get_task_id()),
             startup_state: StartupState {
-                supported: 1_u32 << (Protocol::V1 as u8 - 1),
+                supported: 1_u32 << (Protocol::V2 as u8 - 1),
                 bootrom_crc32: CRC32.checksum(&bootrom().data[..]),
-                buffer_size: BUF_SIZE as u32,
+                max_request_size: MAX_REQUEST_SIZE.try_into().unwrap_lite(),
+                max_response_size: MAX_RESPONSE_SIZE.try_into().unwrap_lite(),
             },
         }
     }
 
     /// Serialize and return a `SprotError::FlowError`
-    pub fn flow_error<'a>(&self, tx_buf: TxMsg<'a>) -> VerifiedTxMsg<'a> {
-        tx_buf.error_rsp(SprotError::FlowError)
+    pub fn flow_error(&self, tx_buf: &mut [u8; MAX_RESPONSE_SIZE]) -> usize {
+        let body = Err(SprotProtocolError::FlowError.into());
+        Response::pack(&body, tx_buf)
     }
 
-    pub fn handle<'a>(
+    pub fn handle(
         &mut self,
-        rx_buf: RxMsg,
-        mut tx_buf: TxMsg<'a>,
+        rx_buf: &[u8],
+        tx_buf: &mut [u8; MAX_RESPONSE_SIZE],
         stats: &mut RotIoStats,
-    ) -> Option<VerifiedTxMsg<'a>> {
-        // Parse the header and validate the CRC
-        let rx_msg = match rx_buf.parse() {
-            Ok(rxmsg) => rxmsg,
-            Err((header_bytes, msgerr)) => {
-                if msgerr == SprotError::NoMessage {
-                    // We were just returning a reply, so clocked out zeros
-                    // from the SP.
-                    return None;
-                }
-                ringbuf_entry!(Trace::ErrWithHeader(msgerr, header_bytes));
+    ) -> usize {
```

Comment:
Nit: you could do the packing in just one place if you write this function as
```rust
    pub fn handle(
        &mut self,
        rx_buf: &[u8],
        tx_buf: &mut [u8; MAX_RESPONSE_SIZE],
        stats: &mut RotIoStats,
    ) -> usize {
        stats.rx_received = stats.rx_received.wrapping_add(1);
        let rsp_body = match Request::unpack(rx_buf) {
            Ok(request) => {
                ringbuf_entry!(Trace::Req {
                    protocol: rx_buf[0],
                    body_type: rx_buf[1]
                });
                self.handle_request(request, stats)
            }
            Err(e) => {
                ringbuf_entry!(Trace::Err(e));
                stats.rx_invalid = stats.rx_invalid.wrapping_add(1);
                Err(e.into())
            }
        };

        Response::pack(&rsp_body, tx_buf)
    }
```

Reply:
Nice! Fixed in [3ddb01b](https://github.com/oxidecomputer/hubris/pull/1279/commits/3ddb01b536d37432e9c62a391d4b708a2630ce23)